src/flb_engine.c

```diff
@@ -207,6 +207,16 @@ static inline int handle_input_event(flb_pipefd_t fd, uint64_t ts,
     return 0;
 }
 
+static inline double calculate_chunk_capacity_percent(struct flb_output_instance *ins)
+{
+    if (ins->total_limit_size == -1) {
```

Comment:
is there a chance that total_limit_size == 0 ? (if so, it will breaks in the next part of the code)

Reply:
There is no chance to be come total_limit_size == 0 due to these lines fill out `-1` value into limit:
https://github.com/fluent/fluent-bit/blob/master/src/flb_output.c#L907-L915